config/initializers/decidim.rb

```diff
@@ -7,21 +7,165 @@
+  config.service_worker_enabled = Rails.application.secrets.decidim[:service_worker_enabled].present?
+
+  # Map and Geocoder configuration
+  #
+  # See Decidim docs at https://docs.decidim.org/en/develop/services/maps.html
+  # for more information about how it works and how to set it up.
+  #
+  # == HERE Maps ==
+  config.maps = {
+    provider: :here,
+    api_key: Rails.application.secrets.maps[:here_api_key],
+    static: { url: "https://image.maps.ls.hereapi.com/mia/1.6/mapview" }
+  }
+  #
+  # == OpenStreetMap (OSM) services ==
+  # To use the OSM map service providers, you will need a service provider for
+  # the following map servers or host all of them yourself:
+  # - A tile server for the dynamic maps
+  #   (https://wiki.openstreetmap.org/wiki/Tile_servers)
+  # - A Nominatim geocoding server for the geocoding functionality
+  #   (https://wiki.openstreetmap.org/wiki/Nominatim)
+  # - A static map server for static map images
+  #   (https://github.com/jperelli/osm-static-maps)
+  #
+  # When used, please read carefully the terms of service for your service
+  # provider.
+  #
+  # config.maps = {
+  #   provider: :osm,
+  #   api_key: Rails.application.secrets.maps[:api_key],
+  #   dynamic: {
+  #     tile_layer: {
+  #       url: "https://tiles.example.org/{z}/{x}/{y}.png?key={apiKey}&{foo}",
+  #       api_key: true,
+  #       foo: "bar=baz",
+  #       attribution: %(
+  #         <a href="https://www.openstreetmap.org/copyright" target="_blank">&copy; OpenStreetMap</a> contributors
+  #       ).strip
+  #       # Translatable attribution:
+  #       # attribution: -> { I18n.t("tile_layer_attribution") }
+  #     }
+  #   },
+  #   static: { url: "https://staticmap.example.org/" },
+  #   geocoding: { host: "nominatim.example.org", use_https: true }
+  # }
+  #
+  # == Combination (OpenStreetMap default + HERE Maps dynamic map tiles) ==
+  # config.maps = {
+  #   provider: :osm,
+  #   api_key: Rails.application.secrets.maps[:api_key],
+  #   dynamic: {
+  #     provider: :here,
+  #     api_key: Rails.application.secrets.maps[:here_api_key]
+  #   },
+  #   static: { url: "https://staticmap.example.org/" },
+  #   geocoding: { host: "nominatim.example.org", use_https: true }
+  # }
+
+  # Geocoder configurations if you want to customize the default geocoding
+  # settings. The maps configuration will manage which geocoding service to use,
+  # so that does not need any additional configuration here. Use this only for
+  # the global geocoder preferences.
+  # config.geocoder = {
+  #   # geocoding service request timeout, in seconds (default 3):
+  #   timeout: 5,
+  #   # set default units to kilometers:
+  #   units: :km,
+  #   # caching (see https://github.com/alexreisner/geocoder#caching for details):
+  #   cache: Redis.new,
+  #   cache_prefix: "..."
+  # }
+  if Rails.application.secrets.maps.present? && Rails.application.secrets.maps[:static_provider].present?
```

Comment:
We need to change the `secrets.yml` maps section. Right now we have something like:

```yaml
  maps:
    here_api_key: <%= ENV["GEOCODER_LOOKUP_API_KEY"] %>
```

And we would need something like:

```yaml
  maps:
    dynamic_provider: <%= Decidim::Env.new("MAPS_DYNAMIC_PROVIDER", ENV["MAPS_PROVIDER"]).to_s %>
    static_provider: <%= Decidim::Env.new("MAPS_STATIC_PROVIDER", ENV["MAPS_PROVIDER"]).to_s  %>
    static_api_key: <%= Decidim::Env.new("MAPS_STATIC_API_KEY", ENV["MAPS_API_KEY"]).to_s %>
    dynamic_api_key: <%= Decidim::Env.new("MAPS_DYNAMIC_API_KEY", ENV["MAPS_API_KEY"]).to_s %>
    dynamic_url: <%= ENV["MAPS_DYNAMIC_URL"] %>
    static_url: <%= ENV["MAPS_STATIC_URL"] %>
    attribution: <%= ENV["MAPS_ATTRIBUTION"].to_json %>
    extra_vars: <%= ENV["MAPS_EXTRA_VARS"].to_json %>
    geocoding_host: <%= ENV["MAPS_GEOCODING_HOST"] %>
```

And as our previous maps configuration in the `decidim.rb` file was something like:

```ruby
  config.maps = {
    provider: :here,
    api_key: Rails.application.secrets.maps[:here_api_key],
    static: { url: "https://image.maps.ls.hereapi.com/mia/1.6/mapview" }
  }
```

And now it's something like:

```ruby
  if Rails.application.secrets.maps.present? && Rails.application.secrets.maps[:static_provider].present?
    static_provider = Rails.application.secrets.maps[:static_provider]
    dynamic_provider = Rails.application.secrets.maps[:dynamic_provider]
    dynamic_url = Rails.application.secrets.maps[:dynamic_url]
    static_url = Rails.application.secrets.maps[:static_url]
    static_url = "https://image.maps.ls.hereapi.com/mia/1.6/mapview" if static_provider == "here" && static_url.blank?
    config.maps = {
      provider: static_provider,
      api_key: Rails.application.secrets.maps[:static_api_key],
      static: { url: static_url },
      dynamic: {
        provider: dynamic_provider,
        api_key: Rails.application.secrets.maps[:dynamic_api_key]
      }
    }
    config.maps[:geocoding] = { host: Rails.application.secrets.maps[:geocoding_host], use_https: true } if Rails.application.secrets.maps[:geocoding_host]
    config.maps[:dynamic][:tile_layer] = {}
    config.maps[:dynamic][:tile_layer][:url] = dynamic_url if dynamic_url
    config.maps[:dynamic][:tile_layer][:attribution] = Rails.application.secrets.maps[:attribution] if Rails.application.secrets.maps[:attribution]
    if Rails.application.secrets.maps[:extra_vars].present?
      vars = URI.decode_www_form(Rails.application.secrets.maps[:extra_vars])
      vars.each do |key, value|
        # perform a naive type conversion
        config.maps[:dynamic][:tile_layer][key] = case value
                                                  when /^true$|^false$/i
                                                    value.downcase == "true"
                                                  when /\A[-+]?\d+\z/
                                                    value.to_i
                                                  else
                                                    value
                                                  end
      end
    end
  end
```

We would need to set the environment variables below:

```
MAPS_PROVIDER=here
MAPS_API_KEY=<same value as GEOCODER_LOOKUP_API_KEY>
```